### PR TITLE
Support secrets

### PR DESCRIPTION
**Pull request by felixb**
This allows to set up framework authentication.

### CODE REVIEW

**Comment by felixb** [comment]
I just rebased this PR on top of your `develop` branch. I guess it's easier to merge to review.
I worked in all your comments (w/o the `require => File[..]`. puppet is doing it's magic to auto-require)

File: manifests/config.pp [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
+  $options                          = {},
+  $env_var                          = {},
+  $manage_logger                    = true,
+  $logger                           = 'logback',
+  $log_dir                          = '/var/log/marathon',
+  $log_filename                     = 'marathon.log',
+  $log_level                        = 'info',
+  $ulimit                           = undef,
+  $secret                           = undef,
+  $mesos_authentication_secret_file = '/etc/marathon/.secret',
```
Comment (JayH5) [inline manifests/config.pp]: Ok.. I've thought about this a bit and can we maybe do:

``` puppet
$mesos_auth_principal   = undef,
$mesos_auth_secret      = undef,
$mesos_auth_secret_file = '/etc/marathon/.secret',
```

And then:

``` puppet
if ($mesos_auth_principal != undef) and ($mesos_auth_secret != undef) {
  $secret_options = {
    'mesos_authentication_principal'   => $mesos_auth_principal,
    'mesos_authentication_secret_file' => $mesos_auth_secret_file,
  }
} else { ...
```

Sorry for chopping and changing... `$mesos_authentication_secret_file` seems a bit long and also the secret is not much good without a principal.

Does this API sound good to you?

**Comment by JayH5** [comment]
Can you explain to me what you mean by "puppet is doing it's magic to auto-require".. As far as I can see the `require` is required although things might work "by accident".

**Comment by felixb** [comment]
Setting all auth options as param is fine. It's a bit tricky to get the overload logic.

```
Autorequires: If Puppet is managing the user or group that owns a file, the file resource will autorequire them. If Puppet is managing any parent directories of a file, the file resource will autorequire them.
```

Source: https://docs.puppetlabs.com/references/latest/type.html#file

**Comment by JayH5** [comment]
Ah thanks. Learn something new every day about Puppet :/

File: manifests/config.pp [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
+  $logger                           = 'logback',
+  $log_dir                          = '/var/log/marathon',
+  $log_filename                     = 'marathon.log',
+  $log_level                        = 'info',
+  $ulimit                           = undef,
+  $mesos_auth_principal             = undef,
+  $mesos_auth_secret                = undef,
+  $mesos_auth_secret_file           = '/etc/marathon/.secret',
+  $java_home                        = undef,
+  $java_opts                        = '-Xmx512m',
```
Comment (JayH5) [inline manifests/config.pp]: Indentation of the `=` a bit messed up here since the parameter names got shorter.

**Comment by felixb** [comment]
I just rebased on current develop and fixed the indenting.

**Comment by JayH5** [comment]
Thanks for your work.

I think we should simplify the Mesos authentication logic and just either completely manage the auth principal/secret or not at all. See this diff: https://gist.github.com/JayH5/57f236f74d917442db2b

**Comment by felixb** [comment]
Sounds good to me. I'll merge your git and update the tests.

**Comment by JayH5** [comment]
lgtm. Can you please add a test where one of the principal/secret is not set and the secrets file and properties are not set?

**Comment by felixb** [comment]
these?

**Comment by JayH5** [comment]
Perfect :) Thank you.

**Comment by JayH5** [comment]
@felixb: Finally got around to releasing version 0.3.0 with your changes :smile:

**Pull request merged**